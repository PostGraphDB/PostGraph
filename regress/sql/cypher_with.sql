/*
 * Copyright (C) 2023-2024 PostGraphDB
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 * Portions Copyright (c) 2020-2023, Apache Software Foundation
 * Portions Copyright (c) 2019-2020, Bitnine Global
 */ 
 
LOAD 'postgraph';
SET search_path TO postgraph;

CREATE GRAPH cypher_with;
USE GRAPH cypher_with;

CYPHER WITH true AS b RETURN b;

CREATE ({i: 1}), ({i: 1, j: 2}), ({i: 2});

MATCH (n) WITH n as a WHERE n.i = 1 RETURN a;

MATCH (n) WITH n as a WHERE n.i = 1 and n.j = 2 RETURN a;


CYPHER WITH generate_series(1, 10) as i RETURN i;

CYPHER WITH generate_series(1, 10, 2) as i RETURN i;

--Error
WITH 1 + 1 RETURN i;


DROP GRAPH cypher_with CASCADE;
