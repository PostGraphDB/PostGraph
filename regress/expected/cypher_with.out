/*
 * Copyright (C) 2023-2024 PostGraphDB
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 * Portions Copyright (c) 2020-2023, Apache Software Foundation
 * Portions Copyright (c) 2019-2020, Bitnine Global
 */ 
 
LOAD 'postgraph';
SET search_path TO postgraph;
CREATE GRAPH cypher_with;
NOTICE:  graph "cypher_with" has been created
 create_graph 
--------------
 
(1 row)

USE GRAPH cypher_with;
 use_graph 
-----------
 
(1 row)

WITH true AS b RETURN b;
ERROR:  syntax error at or near "true"
LINE 1: WITH true AS b RETURN b;
             ^
CREATE ({i: 1}), ({i: 1, j: 2}), ({i: 2});
--
(0 rows)

MATCH (n) WITH n as a WHERE n.i = 1 RETURN a;
                                  a                                   
----------------------------------------------------------------------
 {"id": 281474976710657, "label": "", "properties": {"i": 1}}
 {"id": 281474976710658, "label": "", "properties": {"i": 1, "j": 2}}
(2 rows)

MATCH (n) WITH n as a WHERE n.i = 1 and n.j = 2 RETURN a;
                                  a                                   
----------------------------------------------------------------------
 {"id": 281474976710658, "label": "", "properties": {"i": 1, "j": 2}}
(1 row)

--Error
WITH 1 + 1 RETURN i;
ERROR:  syntax error at or near "1"
LINE 1: WITH 1 + 1 RETURN i;
             ^
DROP GRAPH cypher_with CASCADE;
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to table cypher_with._ag_label_vertex
drop cascades to table cypher_with._ag_label_edge
NOTICE:  graph "cypher_with" has been dropped
 drop_graph 
------------
 
(1 row)

